osu.Game/Database/ArchiveModelManager.cs

```diff
@@ -328,35 +328,51 @@ void rollback()
+                {
+                    using (var write = ContextFactory.GetForWrite())
+                    {
+                        if (!write.IsTransactionLeader) throw new InvalidOperationException($"Ensure there is no parent transaction so errors can correctly be handled by {this}");
+
+                        item.Files = createFileInfos(archive, Files);
+                    }
+                }
+
                 item.Hash = ComputeHash(item, archive);
 
                 await Populate(item, archive, cancellationToken);
 
-                using (var write = ContextFactory.GetForWrite()) // used to share a context for full import. keep in mind this will block all writes.
+                using (var write = ContextFactory.GetForWrite())
                 {
-                    try
-                    {
-                        if (!write.IsTransactionLeader) throw new InvalidOperationException($"Ensure there is no parent transaction so errors can correctly be handled by {this}");
+                    if (!write.IsTransactionLeader) throw new InvalidOperationException($"Ensure there is no parent transaction so errors can correctly be handled by {this}");
 
-                        var existing = CheckForExisting(item);
+                    var existing = CheckForExisting(item);
 
-                        if (existing != null)
+                    if (existing != null)
+                    {
+                        if (CanReuseExisting(existing, item))
                         {
-                            if (CanReuseExisting(existing, item))
-                            {
-                                Undelete(existing);
-                                LogForModel(item, $"Found existing {HumanisedModelName} for {item} (ID {existing.ID}) – skipping import.");
-                                // existing item will be used; rollback new import and exit early.
-                                rollback();
-                                flushEvents(true);
-                                return existing;
-                            }
-
-                            Delete(existing);
-                            ModelStore.PurgeDeletable(s => s.ID == existing.ID);
+                            Undelete(existing);
+                            LogForModel(item, $"Found existing {HumanisedModelName} for {item} (ID {existing.ID}) – skipping import.");
+                            // existing item will be used; rollback new import and exit early.
+                            rollback();
+                            flushEvents(true);
+                            return existing;
                         }
 
+                        Delete(existing);
+                        ModelStore.PurgeDeletable(s => s.ID == existing.ID);
+                    }
+                }
+
+                using (var write = ContextFactory.GetForWrite()) // used to share a context for full import. keep in mind this will block all writes.
```

Comment:
this comment is no longer even true anymore

Reply:
I will remove it later.